README
Geek Code Generator v1.7.3 - Generates your geek code
Copyright (C) 1999-2003 Chris Gushue <chris@blackplasma.net>

-----------------------------------------------------------------------
This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
-----------------------------------------------------------------------

This program will generate a geek code block for you by simply
choosing which codes suit you from the screen. I am not a very good
programmer, I just dabble a little every now and then.

Originally, I started this project in Visual Basic, but decided to use
DJGPP v2 instead, partially because it is free, partially because it is
more portable and partially because Visual Basic sucks :) Since then, I
have totally switched my system to Linux. Currently, my system is
Slackware 4.0 (partially) converted to glibc 2.1.1, but I will be
installing Debian 2.1 very soon.

To use this program, you just run the resulting binary and follow
the directions. Everything SHOULD work, but I haven't tested it
all that much.

If you have any bugs, patches, typos, comments, flames, money, 
postcards, or any other free stuff - feel free to email me
at chris@blackplasma.net - also, please let me know if you modify 
this source, so I can try it myself.

-----------------------------------------------------------------------

Linux notes:

This program was developed using gcc 2.95.1, glibc 2.1.2, and various
other GNU tools, on Debian GNU/Linux (potato), with the 2.2.12 kernel.

To install on Linux (and most likely other UNIX variants):

    make

-----------------------------------------------------------------------

DOS notes:

DJGPP v2 is available at http://www.delorie.com/djgpp/ including
CWSDPMI, which is also released under the GPL. This executable is
required if your OS doesn't provide DPMI services (Windows 9x does).
You have the right to receive the source for CWSDPMI, as well as any
binary updates. These things should also be available at the site
listed above for DJGPP.

To compile on DOS with DJGPP v2, try this:

	gcc -O3 -s -o geekcode.exe *.c

Also, if you have the DJGPP make package instaled, you can probably
just type make, and ignore any errors you might get (at least, I could
do that in my limited testing on my other system).

-----------------------------------------------------------------------

Relevant references:
--------------------
http://www.geekcode.com  - Geek Code website
http://www.gnu.org       - GNU Project and FSF (Free Software Foundation)
http://www.delorie.com   - Home of the DJGPP compiler
http://www.debian.org    - Debian GNU/Linux (the distribution I use)
http://www.slashdot.org  - Geek site extraordinaire
http://www.freshmeat.net - Ultimate Linux software database
http://www.mutt.org      - The best email client
http://www.vim.org       - A nice editor, I recently started using it

-----------------------------------------------------------------------
The Geek Code is copyright (C) 1993, 1994, 1995, 1996 by Robert A.
Hayden. All rights reserved. You are free to distribute this code in
electronic format provided that the file remains unmodified and this
copyright notice remains attached. This copyright prohibits HTMLizing
the code for publication on the web. If you wish to publish abstracts
or portions of the code, contact the author for permission. If you
wish to write an article about the Geek Code, please contact the
author. All creatures not native to Earth are exempt from this
copyright, however, they must prove that they qualify.


CHANGES
Jun 15 1999

- Started working on this Geek Code Generator project

Jun 16 1999

- Managed to get the selection screens finished, all that is left
  to do now is something to convert each numeric variable to a string
  that will be displayed in the final Geek Code.

Aug 04 1999 (version 1.0)

- Finally got around to finishing this project. No DOS binary for now,
  as I am only running Linux on my system (yay!).

Aug 05 1999

- Cleaned up the code a little bit, currently considering how I should
  go about adding the missing features.

Aug 29 1999 (version 1.1)

- Added Unix Type section (page 7a)
- Fixed a few typos

Aug 30 1999 (version 1.2)

- Diff from Fuzzy Logic <fuzzy@mlug.org> to fix some typos and screen
  postitioning.
- Added a #define for the version number (gee, why didn't I do that
  in the first place).
- Added sex type question, and related changes.
- Various other minor changes.

Aug 31 1999 (version 1.3)
- Fixed a couple more typos
- Added little bit of code for each selection screen so that when a
  non-number is typed, the program doesn't start to loop endlessly.
  The compiler generates a warning about using gets(), so is this actually
  a problem, or is there a better way to do it? (Thanks to the few people
  that told me about these bugs).
- Stuff from Kalle Berg <carbe771@student.liu.se>:
  - Added geekcode.h to store the declarations and such
  - Added Win32-specific screen clearing routine (checked at compile time)

Sep 13 1999 (version 1.4)
- Changed the effective screen size to 80x24 (instead of 80x25)
- Changed gets() to use fgets() - thanks Telsa <hobbit@aloss.ukuu.org.uk>

Sep 20 1999 (version 1.5)
- Some minor typo corrections and changes from Fuzzy Logic <fuzzy@mlug.org>
- Added a Quit function (press 0 on any selection screen)

Nov 23 1999
- Added geekcode.spec for compiling on RedHat systems, 
  from Chris Ausbrooks <weed@bucket.pp.ualr.edu>

Feb 15 2000 (version 1.6)
- Added a dynamic page numbering thing,
  from Ingo Hoffmann <lasombra@infolink.com.br>

Feb 16 2000
- Changed the Makefile to actually -use- the CC variable.

Feb 21 2000 (version 1.7)
- I know I wasn't going to put out another version until Geek Code v4.0
  was released, but Ingo Hoffmann sent me a patch to write your Geek Code
  to geekcode.sig as well as display it on the screen.
- My email has also changed to seymour@lazygenes.net
- The URL for this program is now http://seymour.lazygenes.net/geekcode.html

Jun 25 2003 (version 1.7.2)
- The URL for this project has long since changed to http://geekcode.sf.net
  and my email has also been changed to chris@blackplasma.net
- The main change in this version is the inclusion of a Geek Code translator,
  which is supposed to tell you what your Geek Code means.
  Patch supplied by Nikolas Garofil <nikolas.garofil@pandora.be>
- Changed references of Linux to GNU/Linux, as suggested in a Debian bug.

Jun 30 2003 (version 1.7.3)
- Actually made the GNU/Linux changes
- Tiny bugfix from Xavier Santolaria <xavier@santolaria.net>


gc_type.c
#include "geekcode.h"

const struct stuff type[] = {
	{1, "GB",	"Geek of Business"},
	{2, "GC",	"Geek of Classics"},
	{3, "GCA",	"Geek of Commercial Arts"},
	{4, "GCM",	"Geek of Computer Management"},
	{5, "GCS",	"Geek of Computer Science"},
	{6, "GCC",	"Geek of Communications"},
	{7, "GE",	"Geek of Engineering"},
	{8, "GED",	"Geek of Education"},
	{9, "GFA",	"Geek of Fine Arts"},
	{10, "GG",	"Geek of Government"},
	{11, "GH",	"Geek of Humanities"},
	{12, "GIT",	"Geek of Information Technology"},
	{13, "GJ",	"Geek of Jurisprudence (Law)"},
	{14, "GLS",	"Geek of Library Science"},
	{15, "GL",	"Geek of Literature"},
	{16, "GMC",	"Geek of Mass Communications"},
	{17, "GM",	"Geek of Math"},
	{18, "GMD",	"Geek of Medicine"},
	{19, "GMU",	"Geek of Music"},
	{20, "GPA",	"Geek of Performing Arts"},
	{21, "GP",	"Geek of Philosophy"},
	{22, "GS",	"Geek of Science"},
	{23, "GSS",	"Geek of Social Science"},
	{24, "GTW",	"Geek of Technicial Writing"},
	{25, "GO",	"Geek of Other"},
	{26, "GU",	"Geek of Undecided"},
	{27, "G!",	"Geek of No Qualifications"},
	{28, "GAT",	"Geek of All Trades"},
	SENTINEL
};
